src/react/WithQueries.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { Function1 } from 'fp-ts/lib/function';
+import { declareQueries } from './declareQueries';
+import { QueryResult } from '../QueryResult';
+import {
+  EnforceNonEmptyRecord,
+  ObservableQueries,
+  ProductL,
+  ProductP,
+  ProductA
+} from '../util';
+
+type Params<A> = A extends void
+  ? {}
+  : {
+      params: A;
+    };
+
+type Props<R extends ObservableQueries> = {
+  queries: EnforceNonEmptyRecord<R>;
+  render: Function1<QueryResult<ProductL<R>, ProductP<R>>, React.ReactNode>;
+} & Params<ProductA<R>>;
+
+export function WithQueries<P extends ObservableQueries>(props: Props<P>) {
+  const WrappedComponent = React.useMemo(
+    () =>
+      declareQueries(props.queries)(({ queries }) => (
+        <>{props.render(queries)}</>
+      )),
+    [false]
```

Comment:
this should either be `[]` or `Object.values(props.queries)` as described above, right?

Reply:
correct don't know why I put false there :)